Prepare liana v7.0 release (#1325)
wizardsardine/liana: CHANGELOG.md

# Liana daemon and GUI release notes

## 7.0

> This release introduces support for Wizardsardine servers as an optional backend for the Liana GUI

Reviewer: Do we call it Wizardsardine servers now? I thought it was Liana Connect? (also present below)

and Electrum server as an optional node interface for the Liana daemon.

### Breaking changes

> Liana Daemon has a migration for its database to add a new table `transactions` to keep in cache
> transactions to reduce requests to the node. Once Liana GUI or `lianad` run for the first time
> this migration is applied and cannot be reverted.

Reviewer: A bit too many technical details IMO, we should just let the reader now what it means for them.

```suggestion
Running Liana v7 on an existing installation will migrate its database. Once migrated the database won't be compatible with previous versions of Liana. This means you won't be able to open with Liana v6 (or below) any wallet opened or created with Liana v7.
```

### Features

#### Liana daemon / library

> - We added a anti fee snipping protection to the create spend transaction command.

Reviewer: nit, feel free to ignore
```suggestion
- Transactions are now created with fee sniping protection.
```

> - We added optional support for an Electrum server as interface to the bitcoin blockchain instead
>   of Bitcoin Core.

Reviewer: ```suggestion
- You can now remotely connect to your node using Electrum, as an alternative to running Bitcoin Core locally.
```

#### Liana GUI

> - The user does not have anymore to select a network at start up, the launcher looks for a installed
>   network in the following order: `mainnet`, `testnet`, `signet`, `regtest`.

Reviewer: ```suggestion
- The user does not have to select a network at start up anymore. The launcher looks for an installed
  network in the following order: `mainnet`, `testnet`, `signet`, `regtest`.
```

> - The launcher reads the `settings.json` file and provide information about the installed wallet or
>   provide the user a choice to create or add a wallet.

Reviewer: ```suggestion
- The launcher reads the `settings.json` file and provides information about the installed wallet or
  provides the user with a choice to create or add a wallet.
```

- When creating or adding a wallet for the selected network, user has the choice to use his local

>   node or use the Wizardsardine servers has a backend for the Liana GUI.

Reviewer 2: "as a backend"

>   node or use the Wizardsardine servers has a backend for the Liana GUI.

Reviewer: Liana connect vs Ws servers again. Also, the user can use a remote Electrum server so the alternative isn't only his local node. Finally: s/has/as/.

- User can share his hardware wallets xpubs by clicking on "Share xpubs" from the launcher.
- Wallet policy description was added to the installer and in the wallet settings.
- If using the Wizardsardine servers as remote backend, a user can provide access to a wallet to
  an other user by sending him an invitation with the invitee email in the wallet settings.

> - Coldcard and Specter DIY can now be used for any taproot setup.

Reviewer: ```suggestion
- Coldcard and Specter DIY can now be used with taproot descriptors.
```

> - The software version is now in the title bar.

Reviewer: It sounds like it was moved.

```suggestion
- We now display the software version in the title bar.
```

- A `--help` flag was added to display a help menu about the available flags.

> - Bitcoin Core 27.1 is now downloaded if user chooses Liana managed node during install.

Reviewer: ```suggestion
- The Bitcoin Core node version to install for managed local nodes was upgraded to 27.1.  
```

### Fixes

#### Liana daemon / library

> - We check that the change multipath is not hardened while importing descriptor in LianaDescriptor
>   wrapper.

Reviewer: ```suggestion
- We now check the change multipath is not hardened when importing a descriptor (or creating a `LianaDescriptor` struct).
```

> - We add the minimum feerate requested in the error message of the create spend command.

Reviewer: I'm not sure to understand, did we add the requested feerate, or the minimum feerate?

Author: the minimum https://github.com/wizardsardine/liana/pull/1186

#### Liana GUI

> - The rendering of amounts superior to thousands includes now space.

Reviewer: ```suggestion
- The rendering of amounts superior to thousands now includes space.
```

- We detect if the descriptor changed before displaying registration status in the installer.
- We fixed some display issues with overlapping text, notably in the transaction panel.
- On reload, new labels are correctly loaded in existing transaction list in the home panel.

> - In order to handle flaky connection, the whole harware wallet list is reset at each step

Reviewer 2: "hardware"
